android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -2026,48 +1945,41 @@ class QuestionnaireViewModelTest : RobolectricTest() {
                       )
                   }
                 )
-            },
-            Questionnaire.QuestionnaireItemComponent().apply {
-              linkId = "patient-birthDate"
-              addExtension(
-                "http://hl7.org/fhir/uv/sdc/StructureDefinition/sdc-questionnaire-initialExpression",
-                Expression().apply {
-                  language = "text/fhirpath"
-                  expression = "Patient.birthDate"
-                }
-              )
             }
           )
       }
-    val group = Group().apply { id = "112233" }
+    val group = Group().apply { id = "group-1" }
+    val resourceMap: Map<ResourceType?, String> = mapOf()
 
+    // Mandatory get QR from DB, to be used when there's no population resource which triggers an
+    // exception
     coEvery {
       fhirEngine.search<QuestionnaireResponse> {
         filter(QuestionnaireResponse.SUBJECT, { value = group.id })
         filter(QuestionnaireResponse.QUESTIONNAIRE, { value = questionnaire.id })
       }
-    } returns listOf()
+    } returns listOf(QuestionnaireResponse().apply { generateMissingItems(questionnaire) })
 
-    // gets population resources
-    coEvery { fhirEngine.get(group.resourceType, group.id) } returns group
+    // Population resource not found
+    coEvery { fhirEngine.get(group.resourceType, group.id) } throws
+      ResourceNotFoundException(group.resourceType.name, group.id)
 
     val result = runBlocking {
       questionnaireViewModel.getQuestionnaireResponseFromDbOrPopulation(
-        questionnaire,
-        group.id,
-        group.resourceType,
-        questionnaireConfig = questionnaireConfig
+        questionnaire = questionnaire,
+        subjectId = group.id,
+        subjectType = group.resourceType,
+        questionnaireConfig = questionnaireConfig,
+        resourceMap = resourceMap
       )
     }
 
-    assertTrue("Questionnaire Response has no item", result.hasItem())
-
+    assertTrue(result.hasItem())
     coVerify { fhirEngine.get(group.resourceType, group.id) }
-
     val slotPopulationResources = slot<ArrayList<Resource>>()
     coVerify {
       questionnaireViewModel.populateQuestionnaireResponse(any(), capture(slotPopulationResources))
     }
-    assertTrue(slotPopulationResources.captured.isNotEmpty())
+    assertTrue(slotPopulationResources.captured.isEmpty())
```

Comment:
Have we written tests where the captured param is not Empty?

Reply:
yep, check these

- https://github.com/opensrp/fhircore/blob/7fbd1e41d2029edcb1cf5a4d3aada9a4bda968fb/android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt#L1738
- https://github.com/opensrp/fhircore/blob/7fbd1e41d2029edcb1cf5a4d3aada9a4bda968fb/android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt#L1820
- https://github.com/opensrp/fhircore/blob/7fbd1e41d2029edcb1cf5a4d3aada9a4bda968fb/android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt#L1901